File: app/src/main/java/io/github/qwqgong/androidcyaml/RuntimeSnapshot.java
package io.github.qwqgong.androidcyaml;

record RuntimeSnapshot(
        RuntimeState state,
        String detail,
        String dashboardUrl,
        int controllerPort
) {
    static RuntimeSnapshot stopped() {
        return new RuntimeSnapshot(RuntimeState.STOPPED, "VPN 未连接", "", 0);
    }
}

File: app/src/main/java/io/github/qwqgong/androidcyaml/runtime/RuntimeSnapshot.kt
package io.github.qwqgong.androidcyaml

data class RuntimeSnapshot(
    val state: RuntimeState,
    val detail: String,
    val dashboardUrl: String,
    val controllerPort: Int,
) {
    companion object {
        fun stopped(): RuntimeSnapshot = RuntimeSnapshot(RuntimeState.STOPPED, "VPN 未连接", "", 0)
    }
}
